Managed domains

zaibon opened the pull request.

zaibon committed:
make the gateway managed the domain passed in the --domains flag

zaibon committed:
remove the need for redis server to run the cache test

grimpy commented on dns/coredns.go:
```diff
@@ -123,6 +125,11 @@ func (c *Mgr) AddSubdomain(user string, domain string, IPs []net.IP) error {
 
+	// if this is a managed domain and there is already some records, then refuse to modify it
+	if owner.Owner == c.identity && len(zr.Records) > 0 {
+		return fmt.Errorf("the sub-domain %s is already used by someone else: %w", domain, ErrAuth)
```
this means we can not change a subdomain IP even if we made it?

zaibon replied:
Well that's a bit the trick I used here. I do not know who have create the subdomain...
The only thing I know is this subdomain already exists.

zaibon committed:
keep track of who owns subdomain created on top of managed domain

since anyone can created subdomain of top of managed domain we need to
know who has created it so we can prevent other from overwritting it
and/or delete it.

Once the subdomain is released, it becomes available for anyone to take
again.
